website/content/docs/agent/template.mdx

```diff
@@ -116,12 +124,21 @@ The top level `template` block has multiple configurations entries:
   template is rendered. The command will only run if the resulting template changes.
   The command must return within 30s (configurable), and it must have a successful
   exit code. Vault Agent is not a replacement for a process monitor or init system.
+  This is deprecated in favor of the `exec` option.
 - `command_timeout` `(duration: 30s)` - This is the maximum amount of time to
-  wait for the optional command to return.
+  wait for the optional command to return. This is deprecated in favor of the
+  `exec` option.
 - `error_on_missing_key` `(bool: false)` - Exit with an error when accessing
   a struct or map field/key that does notexist. The default behavior will print `<no value>`
   when accessing a field that does not exist. It is highly recommended you set this
   to "true".
+- `exec` `(object: optional)` - The exec block executes a command when the
+  template is rendered. However, the command will only run if the resulting
+  template changes. Please note the exec block is optional, and the command
+  within the exec block must run within a configurable timeout which defaults
+  to 30s. The command must contain a successful exit code. Additionally,
```

Comment:
What happens if we get a non-zero exit code?

Reply:
The error is bubbled up the chain and it's treated as a render error.

Reply:
So Agent will just log an error line essentially?